redpanda/resources/network/resource_network.go

```diff
@@ -153,52 +151,48 @@ func (n *Network) Create(ctx context.Context, request resource.CreateRequest, re
 		response.Diagnostics.AddError("unsupported cluster type", err.Error())
 		return
 	}
-	// TODO add a check to the provider data here to see if region and cloud provider are set
-	// prefer the local value, but accept the provider value if local is
-	// unavailable if neither are set, fail
-
-	op, err := n.NetClient.CreateNetwork(ctx, &controlplanev1beta1.CreateNetworkRequest{
-		Network: &controlplanev1beta1.Network{
-			Name:          model.Name.ValueString(),
-			CidrBlock:     model.CidrBlock.ValueString(),
-			Region:        model.Region.ValueString(),
-			CloudProvider: cloudProvider,
-			NamespaceId:   model.NamespaceID.ValueString(),
-			ClusterType:   clusterType,
+
+	netResp, err := n.CpCl.Network.CreateNetwork(ctx, &controlplanev1beta2.CreateNetworkRequest{
+		Network: &controlplanev1beta2.NetworkCreate{
+			Name:            model.Name.ValueString(),
+			CidrBlock:       model.CidrBlock.ValueString(),
+			Region:          model.Region.ValueString(),
+			CloudProvider:   cloudProvider,
+			ResourceGroupId: model.ResourceGroupID.ValueString(),
+			ClusterType:     clusterType,
```

Comment:
hmm, this is a bit smelly, why is there a `ClusterType` property in the network? Do we mean `NetworkType` perhaps?